test/unit/injector-mutating-webhook.bats

```diff
@@ -53,103 +53,274 @@ load _helpers
   [ "${actual}" = "\"\"" ]
 }
 
-@test "injector/MutatingWebhookConfiguration: namespaceSelector empty by default" {
+@test "injector/MutatingWebhookConfiguration: failurePolicy 'Ignore' by default (deprecated)" {
   cd `chart_dir`
   local actual=$(helm template \
       --show-only templates/injector-mutating-webhook.yaml  \
       --set 'injector.enabled=true' \
+      --set 'injector.webhook=null' \
       --namespace foo \
       . | tee /dev/stderr |
-      yq '.webhooks[0].namespaceSelector' | tee /dev/stderr)
-  [ "${actual}" = "null" ]
+      yq '.webhooks[0].failurePolicy' | tee /dev/stderr)
+  [ "${actual}" = "\"Ignore\"" ]
 }
 
-@test "injector/MutatingWebhookConfiguration: can set namespaceSelector" {
+@test "injector/MutatingWebhookConfiguration: can set failurePolicy (deprecated)" {
   cd `chart_dir`
   local actual=$(helm template \
       --show-only templates/injector-mutating-webhook.yaml  \
       --set 'injector.enabled=true' \
-      --set 'injector.namespaceSelector.matchLabels.injector=true' \
+      --set 'injector.webhook=null' \
```

Comment:
For users upgrading from older versions of the chart, `injector.webhook` typically won't be null, and they could get a behaviour change if they had set `injector.failurePolicy` to `Fail` but don't update anything under `injector.webhook`. Providing a non-null default value is the right thing, but I think we might just need to switch the precedence of the deprecated vs new field, i.e. change from:

```
failurePolicy: {{ ((.Values.injector.webhook)).failurePolicy | default .Values.injector.failurePolicy }}
```

to

```
failurePolicy: {{ .Values.injector.failurePolicy | default ((.Values.injector.webhook)).failurePolicy }}
```

And similar for the other values too. A good test would be that all the new tests you've written still pass without having to use `--set injector.webhook=null` at all.

Reply:
I actually didn't think this would happen. Thank you for letting me know.

My thought process was that: when upgrading, users would copy over their old values to the new fields. Follow the deprecation notes. And make use of the Helm chart as intended.
The other thought process I followed was that users would completely ignore the values.yaml update. Which followed into the new fields needing to be checked using the "`null coalescing operator`". Now if we change this behaviour, this operator might not be necessary anymore as it's not the preferred default.

Won't preferring the old fields over the new ones give any kind of confusion?

Reply:
Yeah I think that seems fair. The disruption is minor and the long term direction is better this way round 👍 thanks for the discussion